Sources/PackageModel/SwiftSDKs/SwiftSDK.swift

```diff
@@ -652,11 +688,24 @@ public struct SwiftSDK: Equatable {
                 pathsConfiguration: .init(sdkRootPath: wasiSysroot)
             )
         }
+
+        #if os(macOS)
+        if let darwinPlatform = targetTriple.darwinPlatform {
+            // the Darwin SDKs are trivially available on macOS
+            var sdk = try? self.systemSwiftSDK(
+                hostSDK.toolset.rootPaths.first,
+                environment: environment,
+                darwinPlatformOverride: darwinPlatform
+            )
+            sdk?.targetTriple = targetTriple
+            return sdk
+        }
+        #endif
+
         return nil
     }
 
     /// Computes the target Swift SDK for the given options.
-    @_spi(SwiftPMInternal)
     public static func deriveTargetSwiftSDK(
```

Comment:
I think it's worth making this non-SPI now so that SourceKit-LSP can use this, instead of its current approach which is to basically duplicate most of this logic (which tends to fall out of sync, case in point this PR). But open to alternatives.

Reply:
Duplicate logic in question: 

https://github.com/swiftlang/sourcekit-lsp/blob/d11c101ce210ad23f917cf06fb13a7d9e243872b/Sources/BuildSystemIntegration/SwiftPMBuildSystem.swift#L229-L248

IMO the above should be replaced with a call to `deriveTargetSwiftSDK`, which would also make it incredibly simple to select iOS as the target in VSCode. See also: https://github.com/swiftlang/sourcekit-lsp/issues/1587#issuecomment-2314111158

Reply:
nevermind, split this change out into #7925